### ISSUE TITLE
Disposable VM created by terminal halts when I start

### ISSUE THREAD

**Issue by lattice0** [QubesOS/qubes-issues#5552]
**Qubes OS version**
4.0.1

**Affected component(s) or functionality**
Template VM customization

**Brief summary**
Did steps in https://www.qubes-os.org/doc/disposablevm-customization/#creating-a-new-disposablevm-template, then changed net vm from sys-firewall to sys-net. When I start any program, like terminal, it appears rapidly and then closes and VM halts

**To Reproduce**
1 - Do exactly what's here (all steps) https://www.qubes-os.org/doc/disposablevm-customization/#creating-a-new-disposablevm-template
2 - Change from sys-firewall to sys-net
3 - Launch VM (any app)
4 - App appears and suddenly closes and VM halts
5 - Can't delete disposable VM template in qubes gui menu, it says "it is used as: Global property default_dispvm"

**Comment by lucaszanella** [comment]
Just figured out how to delete it. Information on https://www.qubes-os.org/doc/disposablevm-customization/#deleting-disposablevms is not correct, I guess. I tried and it didn't work. I had to go back to the top to see

qubes-prefs default_dispvm custom-disposablevm-template

so I just setted another VM to be the default, then delete worked.

Anyways, the step "change from sys-firewall to sys-net" it not important, the problem happens when I don't do that as well, just created a brand new one and based on other template, and the same happened.

**Comment by andrewdavidwong** [comment]
What exactly do you mean by "2 - Change from sys-firewall to sys-net"?

Also, which VM are you launching in step 3?

**Comment by lucaszanella** [comment]
I meant changing the networking VM of the disposable VM from sys-firewall to sys-net, but I just confirmed that this step is not necessary

I launched, in step 3, the disposable VM. I launched the terminal but any app will cause the same effect

-------- Mensagem Original --------
Ativo 3 de jan de 2020 00:02, Andrew David Wong escreveu:

**Comment by SaswatPadhi** [comment]
This is a known issue (#2581) with `gnome-terminal`. However, `xterm` works fine.

**Comment by andrewdavidwong** [comment]
@lucaszanella:

There is a difference between a [DisposableVM](https://www.qubes-os.org/doc/glossary/#disposablevm) and a [DisposableVM Template](https://www.qubes-os.org/doc/glossary/#disposablevm-template). Based on your issue description, it sounds like you may not be aware of this distinction.

You followed the [instructions for creating a DisposableVM Template](https://www.qubes-os.org/doc/disposablevm-customization/#creating-a-new-disposablevm-template).

After following those instructions, are you attempting to launch the **DisposableVM Template** you just created or a **DisposableVM** _based on_ that **DisposableVM Template**?

If you're trying to start a **DisposableVM**, make sure it's based on the correct **DisposableVM Template**, since you now have more than one.

**Comment by lucaszanella** [comment]
@andrewdavidwong I just followed the steps to create a disposable template and then launched it through the GUI. It appears just like any other VM. I thougth it simply created an anonymous dispvm based on the template whenever I launched. Isn't it true? Indeed, Firefox works fine, but Files, Gnome Terminal, Text Editor and possibly other things don't start. But now I understand that it's not a bug
